solr/packaging/powershell-tests/Zk.Tests.ps1

```diff
@@ -0,0 +1,208 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+<#
+  Pester tests for Solr zk command
+  Ported from test/test_zk.bats
+#>
+
+BeforeAll {
+  # Get the Solr installation directory from environment variable
+  $script:SolrTip = $env:SOLR_TIP
+
+  # Note: $SolrTip cannot be used directly in -Skip statements in tests, use $env:SOLR_TIP instead
+  if (-not $SolrTip) {
+    throw "SOLR_TIP environment variable is not set"
+  }
+
+  # Determine the Solr executable based on OS
+  $script:SolrCmd = Join-Path $SolrTip "bin\solr.cmd"
+
+  if (-not (Test-Path $SolrCmd)) {
+    throw "Solr executable not found at: $SolrCmd"
+  }
+
+  Write-Host "Using Solr installation at: $SolrTip"
+  Write-Host "Using Solr command: $SolrCmd"
+
+  # Get ZooKeeper port from environment or default
+  $script:ZK_PORT = if ($env:ZK_PORT) { $env:ZK_PORT } else { "9983" }
+  $script:SOLR_PORT = if ($env:SOLR_PORT) { $env:SOLR_PORT } else { "8983" }
+
+  function Test-CommandOutput {
+    param(
+      [string[]]$Arguments,
+      [string]$TestName
+    )
+
+    Write-Host "Testing: $TestName"
+    Write-Host "Running: $SolrCmd $($Arguments -join ' ')"
+
+    $output = & $SolrCmd @Arguments 2>&1
+    $outputStr = $output | Out-String
+
+    Write-Host "Exit Code: $LASTEXITCODE"
+    if ($outputStr.Length -gt 0) {
+      Write-Host "Output (first 500 chars): $($outputStr.Substring(0, [Math]::Min(500, $outputStr.Length)))"
+    } else {
+      Write-Host "WARNING: Output is empty!"
+    }
+
+    return $outputStr
+  }
+}
+
+Describe "Solr Zk Command" {
+  Context "Help commands" {
+    It "short help" {
+      $output = Test-CommandOutput @("zk", "ls", "-h") "zk ls -h"
+      $output | Should -Match "usage: bin/solr zk"
+    }
+
+    It "short help is inferred" {
+      $output = Test-CommandOutput @("zk", "ls") "zk ls"
+      $output | Should -Match "usage: bin/solr zk"
+    }
+
+    It "long help" {
+      $output = Test-CommandOutput @("zk", "-h") "zk -h"
+      $output | Should -Match "bin/solr zk ls"
+      $output | Should -Match "bin/solr zk updateacls"
+      $output | Should -Match "Pass --help or -h after any COMMAND"
+    }
+
+    It "running subcommands with zk is prevented" {
+      $output = Test-CommandOutput @("ls", "/", "-z", "localhost:$ZK_PORT") "ls / -z localhost:$ZK_PORT"
+      $output | Should -Match "You must invoke this subcommand using the zk command"
+    }
+  }
+
+  Context "Zk operations (requires running Solr with ZooKeeper)" -Skip:(-not (Test-Path (Join-Path $env:SOLR_TIP "server\solr"))) {
+    BeforeAll {
+      # Check if Solr is running by trying to connect to the port
+      $solrRunning = $false
+      try {
+        $response = Invoke-WebRequest -Uri "http://localhost:$SOLR_PORT/solr/" -UseBasicParsing -ErrorAction SilentlyContinue
+        if ($response.StatusCode -eq 200) {
+          $solrRunning = $true
+        }
+      } catch {
```

Comment:
This block always fails, leading to the 9 tests being skipped. Is Solr ever started during the test?